js/components/MilkdropWindow/index.js

```diff
@@ -164,12 +178,27 @@ async function loadInitialPreset() {
         throw new Error(response.statusText);
       }
       const responseBody = await response.json();
-      return responseBody;
+      const presetName = presetNameFromURL(presetUrl);
+      return { [presetName]: responseBody };
     } catch (e) {
       console.error(e);
       alert(`Failed to load MilkDrop preset from ${presetUrl}`);
       return null;
     }
+  } else if (milkdropPresetUrl) {
+    try {
+      const response = await fetch(milkdropPresetUrl);
+      if (!response.ok) {
+        throw new Error(response.statusText);
+      }
+      const fileBlob = await response.blob();
+      const presetName = presetNameFromURL(milkdropPresetUrl);
+      return { [presetName]: { file: fileBlob } };
+    } catch (e) {
```

Comment:
I'd rather this try/catch were narrower, even if it means having multiple of them. Generally, I think we should know exactly what kinds of errors we expect to be catching for each try/catch. Otherwise we end up swallowing errors which really should result in an exception.

Reply:
Feels odd specifying catching of `Error` at the end, but its because thats what we are throwing when the response is bad.  I guess we could create a subclass to throw for that so its more explicit?

Reply:
As an alternative to a subclass, you could provide a property on the error like Node.js does with `error.code`
Also, shouldn't the `TypeError` case be handled by re-throwing? Assuming it means you've used the interface wrong.